solutions/src/main/java/ru/sberbank/school/task02/ExtendedConverter.java

```diff
@@ -0,0 +1,83 @@
+package ru.sberbank.school.task02;
+
+import ru.sberbank.school.task02.exception.ConverterConfigurationException;
+import ru.sberbank.school.task02.util.Beneficiary;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+
+public class ExtendedConverter extends Converter implements ExtendedFxConversionService {
+
+    private ExternalQuotesService externalQuotesService;
+
+    public ExtendedConverter(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
+        this.externalQuotesService = externalQuotesService;
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol, BigDecimal amount, Beneficiary beneficiary) {
+        List<Quote> quotes = externalQuotesService.getQuotes(symbol);
+
+        checkData(quotes, amount, beneficiary);
+
+        Quote quote = null;
+        Quote reserveQuote = null;
+
+        for (Quote currQuote : quotes) {
+            BigDecimal price = getPrice(operation, currQuote);
+            BigDecimal usd = amount.divide(price, 10, BigDecimal.ROUND_HALF_UP);
+
+            if (currQuote.getVolumeSize().compareTo(usd) > 0 || currQuote.isInfinity()) {
+                Quote quote3 = searchQuote(usd, quotes);
+                if (currQuote == quote3) {
+                    if (quote == null)
```

Comment:
&#39;if&#39; construct must use &#39;{}&#39;s.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1175050902>*